### ISSUE TITLE
If input is empty and user submits form using ENTER key, server receives input mask instead of empty value

### ISSUE THREAD

**Issue by ostrolucky** [RobinHerbots/Inputmask#624]
jquery.inputmask does good job removing mask from input before submitting when user uses his mouse (because it removes mask from non-focused empty inputs), but fails to do it when user has focus on empty masked input and sends form using ENTER key

**Comment by ostrolucky** [comment]
Solution to this problem is apparently setting autoUnmask to true. Any particular reason why is this not enabled by default?

**Comment by RobinHerbots** [comment]
@gadelat , 

The above works because autoUnmask in combination of removeMaskOnSubmit will remove the mask on submit.  

It it not enabled by default because some people also want the masked value when getting/submitting a value

But can you have a test with version 3.1.1  (so without the autoUnmask)

**Comment by RobinHerbots** [comment]
@gadelat , 

I reupdated version 3.1.1 because I was a bit too fast, ... trying to do too many things at the same time.

If you already tried, ... try again

**Comment by ostrolucky** [comment]
Nope, still there

**Comment by ostrolucky** [comment]
Ok, tested some more now.  Let me clarify.  It kinda works - if value is not set by default.

If you focus input below and submit form using enter, it works as it should (server receives empty value)

``` html
<input name="date" data-inputmask="'alias': 'date'" />
```

If you focus input below, remove default value and submit form using enter, it doesn't work as it should (server receives dd/mm/yyyy)

``` html
<input name="date" data-inputmask="'alias': 'date'" value="08/12/1989" />
```

**Comment by RobinHerbots** [comment]
@gadelat , 

Ok I will have a look at it.

**Comment by RobinHerbots** [comment]
@gadelat , 

Fixed in version 3.1.4
